src/main/scala/repo/Repository.scala

```diff
@@ -0,0 +1,23 @@
+package repo
+
+import cats.data.State
+import models.Domain.{Customer, CustomerId}
+
+object Repository {
+
+  trait CustomerRepository {
+    def add(customer: Customer): Unit
```

Comment:
assume that `add` may result in error, u can use either `IO.raiseError` or `IO[Either]` and then in gateway layer translate domain error to http 500